test/consistent-function-scoping.js

```diff
@@ -362,6 +368,25 @@ ruleTester.run('consistent-function-scoping', rule, {
 				}
 			`,
 			errors: [functionError]
+		},
+		// React Hooks
+		{
+			code: outdent`
+				useEffect(() => {
+					function foo() {
+						function bar() {
+						}
+					}
+				}, [])
+			`,
+			errors: [
+				{
+					ruleId: 'consistent-function-scoping',
+					messageId: 'FunctionDeclaration',
+					column: 3,
+					line: 3
+				}
+			]
```

Comment:
Todo: use function name to test after merge https://github.com/sindresorhus/eslint-plugin-unicorn/pull/589